pkg/operation/hybridbotanist/controlplane.go

```diff
@@ -497,3 +432,135 @@ func (b *HybridBotanist) DeployKubeScheduler() error {
 
 	return b.ApplyChartSeed(filepath.Join(chartPathControlPlane, gardencorev1alpha1.DeploymentNameKubeScheduler), b.Shoot.SeedNamespace, gardencorev1alpha1.DeploymentNameKubeScheduler, values, nil)
 }
+
+// DeployBackupEntryInGarden deploys the BackupEntry resource in garden.
+func (b *HybridBotanist) DeployBackupEntryInGarden(ctx context.Context) error {
+	var (
+		name        = common.GenerateBackupEntryName(b.Shoot.Info.Status.TechnicalID, b.Shoot.Info.Status.UID)
+		backupEntry = &gardencorev1alpha1.BackupEntry{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: b.Shoot.Info.Namespace,
+			},
+		}
+		bucketName string
+		seedName   *string
+	)
+
+	if err := b.K8sGardenClient.Client().Get(ctx, kutil.Key(b.Shoot.Info.Namespace, name), backupEntry); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		// If backupEntry doesn't already exists, we have to assign backupBucket to backupEntry.
+		cloudProfile := b.Seed.CloudProfile
+		if cloudProfile.Spec.Backup.Region != nil && len(*cloudProfile.Spec.Backup.Region) != 0 {
+			bucketName = common.GenerateBackupBucketName(cloudProfile.Name, *cloudProfile.Spec.Backup.Region, cloudProfile.UID)
+		} else {
+			bucketName = common.GenerateBackupBucketName(cloudProfile.Name, b.Seed.Info.Spec.Cloud.Region, cloudProfile.UID)
+		}
+		seedName = &b.Seed.Info.Name
+	} else {
+		bucketName = backupEntry.Spec.BucketName
+		seedName = backupEntry.Spec.Seed
+	}
+	ownerRef := metav1.NewControllerRef(b.Shoot.Info, gardenv1beta1.SchemeGroupVersion.WithKind("Shoot"))
+
+	return kutil.CreateOrUpdate(ctx, b.K8sGardenClient.Client(), backupEntry, func() error {
+		backupEntry.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+		backupEntry.Spec.BucketName = bucketName
+		backupEntry.Spec.Seed = seedName
+		return nil
+	})
+}
+
+// DeployETCD deploys two etcd clusters via StatefulSets. The first etcd cluster (called 'main') is used for all the
+// data the Shoot Kubernetes cluster needs to store, whereas the second etcd luster (called 'events') is only used to
+// store the events data. The objectstore is also set up to store the backups.
+func (b *HybridBotanist) DeployETCD(ctx context.Context) error {
+	var (
+		backupInfraName      = common.GenerateBackupInfrastructureName(b.Shoot.Info.Status.TechnicalID, b.Shoot.Info.Status.UID)
+		lastSnapshotRevision int64
+	)
+	backupInfra := &gardenv1beta1.BackupInfrastructure{}
+	err := b.K8sGardenClient.Client().Get(ctx, kutil.Key(b.Shoot.Info.Namespace, backupInfraName), backupInfra)
+	if err == nil {
```

Comment:
First handle the error, and then the "good" case

Reply:
Sure